src/main/java/duke/command/Action.java

```diff
@@ -0,0 +1,70 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidFormatException;
+import duke.exception.InvalidIntegerException;
+import duke.exception.InvalidTaskNumberException;
+import duke.Storage;
+import duke.TaskList;
+import duke.tasks.Task;
+import duke.Ui;
+
+/**
+ * Class that handles the Done and Delete command
+ */
+public class Action extends Command {
+
+    private enum Type {
+        DONE,
+        DELETE
+    }
+    
+    private final Type type;
+    private final String[] words;
+
+    public Action(int type, String[] words) {
+        this.type = type == 0 ? Type.DONE : Type.DELETE;
+        this.words = words;
+    }
+
+    @Override
```

Comment:
You may want to ass a javadoc for this public method since it is quite important~